shell/common/rasterizer_unittests.cc

```diff
@@ -830,4 +831,152 @@ TEST(RasterizerTest,
   latch.Wait();
 }
 
+TEST(RasterizerTest, presentationTimeSetWhenVsyncTargetInFuture) {
+  std::string test_name =
+      ::testing::UnitTest::GetInstance()->current_test_info()->name();
+  ThreadHost thread_host("io.flutter.test." + test_name + ".",
+                         ThreadHost::Type::Platform | ThreadHost::Type::RASTER |
+                             ThreadHost::Type::IO | ThreadHost::Type::UI);
+  TaskRunners task_runners("test", thread_host.platform_thread->GetTaskRunner(),
+                           thread_host.raster_thread->GetTaskRunner(),
+                           thread_host.ui_thread->GetTaskRunner(),
+                           thread_host.io_thread->GetTaskRunner());
+  MockDelegate delegate;
```

Comment:
nit: use `NiceMock<MockDelegate>` (`using ::testing::NiceMock`) so this doesn't spam output about unused calls, here and elsewhere.

Reply:
nice! https://github.com/flutter/flutter/issues/105631, i think we want to use it everywhere in this file.

Reply:
I'm updating the rest of the file in https://github.com/flutter/engine/pull/33890 which is currently stuck on Fuchsia weirdness.